### PR TITLE
Fixed public API for file get range diff

### PR DESCRIPTION
**Pull request by gapra-msft**
Public API of file get range diff now returns a ShareFileRangeList to follow PageBlob API instead of existing file get range API. There is no reason to use PagedFlux since the response is not paginated.

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

**Comment by gapra-msft** [comment]
@alzimmermsft Can you take a look for API feedback?

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java [hunk 1]
```diff
@@ -1636,11 +1637,12 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
      * snapshot, as long as the snapshot specified by previousSnapshot is the older of the two.
      * @return {@link ShareFileRange ranges} in the files that satisfy the requirements
      */
-    public PagedFlux<ShareFileRange> listRangesDiff(String previousSnapshot) {
+    public Mono<ShareFileRangeList> listRangesDiff(String previousSnapshot) {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java:1640]: From a functional stand-point this change is fine. Generally, we have erred on using `PagedFlux`/`PagedIterable` on list responses so if the service changed in the future to be pagable we wouldn't require an API change in the SDK.

https://azure.github.io/azure-sdk/java_design.html#java-pagination-pagedflux

Comment (seanmcc-msft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java:1640]: We have a Mono<> for PageBlobClient.GetPageRange() and .GetPageRangeDiff(), and this API isn't currently Pagable, I think it's important that we be internally consistent..

Comment (kasobol-msft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java:1640]: I believe the chance this api going pageable is small.

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java:1640]: Discussed offline. This is a list response but there are 2 different lists returned by the service - ClearRange and Range, so going with a Mono makes sense here.

Comment (alzimmermsft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java:1640]: Didn't realize the response type was two disparate lists, given that `Mono` makes more sense.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**